ci/test/04_install.sh

```diff
@@ -42,7 +42,9 @@ if [ -z "$DANGER_RUN_CI_ON_HOST" ]; then
 
   if [ -n "${RESTART_CI_DOCKER_BEFORE_RUN}" ] ; then
     echo "Restart docker before run to stop and clear all containers started with --rm"
-    systemctl restart docker
+    podman container kill --all  # Similar to "systemctl restart docker"
+    echo "Prune all dangling images"
+    docker image prune --force
```

Comment:
Would it be prudent to use the [`--filter` option](https://docs.docker.com/engine/reference/commandline/image_prune/#filter) here to only prune images created by the CI? This may be especially helpful to people running the CI locally/unsandboxed. We'd just need to add a `--label "bitcoin-ci"` argument to `docker build`.

```suggestion
    docker image prune --force --filter label=bitcoin-ci
```

Reply:
> Do dangling image (layer)s have a label?

Yeah they do, just tested it to confirm.

> guarded via RESTART_CI_DOCKER_BEFORE_RUN

Fair enough, looks like this is unlikely to be hit locally. I don't think renaming to `DANGER_RESTART_CI_DOCKER_BEFORE_RUN` is an improvement, so happy to keep as is. Just thought adding a label and only applying batch operations to our own images is a good practice, e.g. if in the future we remove the `RESTART_CI_DOCKER_BEFORE_RUN` condition, given that it's almost no overhead. But up to you, it's not big deal and I don't have a strong opinion.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/27793